nx/guides/modoc_agg_nx_01.livemd

```diff
@@ -0,0 +1,748 @@
+  first_row,
+  first_column,
+  reverse_index
+}
+```
+
+## Aggregations operations on tensors
+
+You can apply [aggregation functions](https://hexdocs.pm/nx/Nx.html#functions-aggregates) on any tensor. The function can be applied to the tensor as a whole or to a subsection of the tensor taken in an axis-wise fashion.
+
+As a first example, let's take a 2D tensor (aka matrix) of shape (2,3). Notice that we can _name_ the axes. The elements of the tensor `t` are the `t[i][j]` when $i$ is the row, and $j$ the column.
+
+```elixir
+m = Nx.tensor([[1, 2, 3], [4, 5, 6]], names: [:x, :y])
+```
+
+You can get the maximum number in a tensor with [Nx.reduce_max(matrix)](https://hexdocs.pm/nx/Nx.html#reduce_max/2), returning a scalar (0D tensor). With a reduction, we loose a dimension per axis reduced, and since we did globally, we loose all the dimensions. It should return `Nx.Tensor(6)` here.
+
+We can get the maximum number for each _row_ with `Nx.reduce_max(matrix, axes: [:y])`, returning a vector (1D tensor) of length 2. Why `:y`? Because for each row, we reduce along the `:y` axis.
+
+We can get the maximum number for each _column_ with `Nx.reduce_max(matrix, axes: [:x])`, returning a vector (1D tensor) of length 3. For each column, we reduce along the axis `:x`.
+
+```elixir
+max = Nx.reduce_max(m)
+max_x = Nx.reduce_max(m, axes: [:x])
+max_y = Nx.reduce_max(m, axes: [:y])
+{max, max_x, max_y, m}
+```
+
+Let's consider another example with the aggregate [Nx.weighted_mean](https://hexdocs.pm/nx/Nx.html#weighted_mean/3). It supports full-tensor and per axis operations. Let's check how the weighted mean aggregate of a matrix is computed with a 2D tensor of shape (2,2) (aka as "square matrix").
+
+```elixir
+m = ~M(1 2
+  3 4)
+```
+
+We compute the full -tensor aggregate. The calculations are developped below. We calculate an "array product" (aka [Hadamard product](<https://en.wikipedia.org/wiki/Hadamard_product_(matrices)#:~:text=In%20mathematics%2C%20the%20Hadamard%20product,elements%20i%2C%20j%20of%20the>), a terms by terms product) of our tensor with the tensor of weights, then sum all the elements and divide by the sum of the weights.
+
+```elixir
+w = ~M[4 5
+  4 5]
+
+nx_w_avg = Nx.weighted_mean(m, w)
+
+man_w_avg = (1 * 4 + 2 * 5 + 3 * 4 + 4 * 5) / (4 + 5 + 4 + 5)
+# 46/18
+
+%{
+  nx_weighted_avg: Nx.to_number(nx_w_avg),
+  manual_weighted_avg: man_w_avg
+}
+```
+
+The weighted mean can be run per axis. Let's compute it along the **first** axis (`axes: [0]`) with the same weights per axis: you calculate "by column", so you aggregate/reduce along the first axis:
+
+```elixir
+w = ~M(4 5
+  4 5)
+
+w_avg_x = Nx.weighted_mean(m, w, axes: [0])
+
+man_w_avg_x = [(1 * 4 + 3 * 4) / (4 + 4), (2 * 5 + 4 * 5) / (5 + 5)]
+
+{
+  w_avg_x,
+  man_w_avg_x
+}
+```
+
+We calculate weighted mean of a square matrix along the **second** axis (`axes: [1]`): you calculate per row, so you aggregate/reduce along the second axis.
+
+```elixir
+w = ~M(4 5
+  5 4)
```

Comment:
I'd use totally different values instead of just changing the order

Reply:
Done!